aiida/cmdline/utils/echo.py

```diff
@@ -169,7 +169,8 @@ def echo_dictionary(dictionary, fmt='json+date'):
     :param dictionary: the dictionary
     :param fmt: the format to use for printing, valid options: ['json+data']
     """
-    valid_formats_table = {'json+date': _format_dictionary_json_date}
+    from aiida.orm.utils.serialize import serialize
+    valid_formats_table = {'json+date': _format_dictionary_json_date, 'yaml': serialize}
```

Comment:
Could you move this as a module variable (capitalised, e.g. called `VALID_DICT_FORMATS_MAPPING`) and then reuse it here, but also when you do `type=click.Choice(["json+date", "yaml"])` you replace this with `VALID_DICT_FORMATS_MAPPING.keys()`?

Also, could you add the `format` option to the other functions calling the function (by grepping `json+date` we found at least `verdi data array` and `verdi data dict`